users which are created by the web API are not added to the DB anymore

- Operating System (Linux/Mac/Windows/iOS/Android): Linux
- Expected behavior: when an email account is created via the web route, it gets added to the database and pruned later when it expires.
- Actual behavior: the account is not added to the database, so it never gets deleted. If you check `mailadm list-users`, it appears only with a `[created in mailcow]` addition, not with the token which actually created it.
- Steps to reproduce the problem:
  - Setup a development environment: https://mailadm.readthedocs.io/en/latest/#setup-development-environment
  - run `git checkout missytake/fix-user-not-in-db`
  - get a `MAILCOW_TOKEN` from dc.develcow.de: https://mailadm.readthedocs.io/en/latest/#mailcow-token
  - Run `MAILCOW_TOKEN=XXX pytest tests/test_web.py::test_user_in_db`
- Logs:

```
pytest tests/test_web.py::test_user_in_db
============================================================================================================ test session starts ============================================================================================================
platform linux -- Python 3.10.6, pytest-7.2.0, pluggy-1.0.0
Deltachat core=v1.101.0 sqlite=3.37.2 journal_mode=wal
rootdir: /home/user/git/mailadm
plugins: timeout-2.1.0, deltachat-1.101.0, xdist-3.0.2
collected 1 item                                                                                                                                                                                                                            

tests/test_web.py F                                                                                                                                                                                                                   [100%]

================================================================================================================= FAILURES ==================================================================================================================
______________________________________________________________________________________________________________ test_user_in_db ______________________________________________________________________________________________________________

db = <mailadm.db.DB object at 0x7fae26071930>, mailcow = <mailadm.mailcow.MailcowConnection object at 0x7fae26072500>

    def test_user_in_db(db, mailcow):
        with db.write_transaction() as conn:
            token = conn.add_token("pytest:web", expiry="1w", token="1w_7wDioPeeXyZx96v", prefix="")
        app = create_app_from_db_path(db.path)
        app.debug = True
        app = app.test_client()
    
        r = app.post('/?t=' + token.token)
        assert r.status_code == 200
        assert r.json["password"]
        addr = r.json["email"]
    
        assert mailcow.get_user(addr)
        with db.read_connection() as conn:
>           assert conn.get_user_by_addr(addr)

tests/test_web.py:77: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <mailadm.conn.Connection object at 0x7fae26073ee0>, addr = 'jsnfk@x.testrun.org'

    def get_user_by_addr(self, addr):
        q = UserInfo._select_user_columns + "WHERE addr = ?"
        args = self._sqlconn.execute(q, (addr, )).fetchone()
>       return UserInfo(*args)
E       TypeError: mailadm.conn.UserInfo() argument after * must be an iterable, not NoneType

src/mailadm/conn.py:226: TypeError
----------------------------------------------------------------------------------------------------------- Captured stdout setup -----------------------------------------------------------------------------------------------------------
DB: Creating tables /tmp/pytest-of-user/pytest-107/test_user_in_db0/base/mailadm.db
----------------------------------------------------------------------------------------------------------- Captured stdout call ------------------------------------------------------------------------------------------------------------
added token 'pytest:web'
added addr 'jsnfk@x.testrun.org' with token 'pytest:web'
========================================================================================================== short test summary info ==========================================================================================================
FAILED tests/test_web.py::test_user_in_db - TypeError: mailadm.conn.UserInfo() argument after * must be an iterable, not NoneType
============================================================================================================= 1 failed in 1.64s =============================================================================================================
```

--- a/src/mailadm/db.py
+++ b/src/mailadm/db.py
@@ -18,8 +18,9 @@ def get_db_path():
 
 
 class DB:
-    def __init__(self, path, autoinit=True):
+    def __init__(self, path, autoinit=True, debug=False):
         self.path = path
+        self.debug = debug
         self.ensure_tables()
 
     def _get_connection(self, write=False, transaction=False, closing=False):
@@ -37,6 +38,8 @@ class DB:
             isolation_level=None if transaction else "DEFERRED",
             uri=True,
         )
+        if self.debug:
+            sqlconn.set_trace_callback(print)
 
         # Enable Write-Ahead Logging to avoid readers blocking writers and vice versa.
         if write:

--- a/src/mailadm/web.py
+++ b/src/mailadm/web.py
@@ -24,7 +24,7 @@ def create_app_from_db(db):
             return jsonify(type="error", status_code=403,
                            reason="?t (token) parameter not specified"), 403
 
-        with db.write_connection() as conn:
+        with db.write_transaction() as conn:
             token_info = conn.get_tokeninfo_by_token(token)
             if token_info is None:
                 return jsonify(type="error", status_code=403,

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -176,7 +176,7 @@ def make_db(monkeypatch, mailcow_auth, mailcow_endpoint):
     def make_db(basedir, init=True):
         basedir = Path(str(basedir))
         db_path = basedir.joinpath("mailadm.db")
-        db = mailadm.db.DB(db_path)
+        db = mailadm.db.DB(db_path, debug=True)
         if init:
             db.init_config(
                 mail_domain="x.testrun.org",

--- a/tests/test_conn.py
+++ b/tests/test_conn.py
@@ -19,6 +19,22 @@ def test_token_twice(conn):
         conn.add_token("burner2", expiry="1w", token="1w_7wDioPeeXyZx96v3", prefix="xp")
 
 
+def test_add_del_user(conn, mailcow):
+    token = conn.add_token("pytest:burner1", expiry="1w", token="1w_7wDioPeeXyZx96v3", prefix="pp")
+    usecount = conn.get_tokeninfo_by_name(token.name).usecount
+    addr = conn.add_email_account_tries(token, tries=10).addr
+    assert usecount + 1 == conn.get_tokeninfo_by_name(token.name).usecount
+
+    assert mailcow.get_user(addr)
+    assert conn.get_user_by_addr(addr)
+
+    conn.delete_email_account(addr)
+
+    with pytest.raises(TypeError):
+        conn.get_user_by_addr(addr)
+    assert not mailcow.get_user(addr)
+
+
 def test_token_info(conn):
     conn.add_token("burner1", expiry="1w", token="1w_7wDioPeeXyZx96v3", prefix="pp")
     conn.add_token("burner2", expiry="10w", token="10w_7wDioPeeXyZx96v3", prefix="xp")

--- a/tests/test_web.py
+++ b/tests/test_web.py
@@ -60,6 +60,26 @@ def test_env(db, monkeypatch):
     assert app.db.path == db.path
 
 
+def test_user_in_db(db, mailcow):
+    with db.write_transaction() as conn:
+        token = conn.add_token("pytest:web", expiry="1w", token="1w_7wDioPeeXyZx96v", prefix="")
+    app = create_app_from_db_path(db.path)
+    app.debug = True
+    app = app.test_client()
+
+    r = app.post('/?t=' + token.token)
+    assert r.status_code == 200
+    assert r.json["password"]
+    addr = r.json["email"]
+
+    assert mailcow.get_user(addr)
+    with db.read_connection() as conn:
+        assert conn.get_user_by_addr(addr)
+
+    with db.write_transaction() as conn:
+        conn.delete_email_account(addr)
+
+
 # we used to allow setting the username/password through the web
 # but the code has been removed, let's keep the test around
 def xxxtest_new_user_usermod(db):

